### PR TITLE
Add github actions to Build on push, pull_request and Release on tag push

### PR DESCRIPTION
**Pull request by shaileshpandit**
- Creates release on tag push
- Creates and adds pre-compiled.zip as release artifact

### CODE REVIEW

File: build/media-autobuild_suite.ini [hunk 1]
```diff
@@ -41,12 +41,12 @@ jo=2
 dssim=2
 avs2=2
 CC=2
-cores=4
+cores=2
```
Comment (shaileshpandit) [inline build/media-autobuild_suite.ini:44]: Github runners only has 2 cores, for local builds one has to change this for now, we can maintain 2 separate configs if necessary.

File: build/media-autobuild_suite.ini [hunk 2]
```diff
@@ -41,12 +41,12 @@ jo=2
...
 deleteSource=2
 strip=1
 pack=2
 logging=1
 updateSuite=1
 timeStamp=1
-ccache=2
-noMintty=2
+ccache=1
+noMintty=1
```
Comment (shaileshpandit) [inline build/media-autobuild_suite.ini:52]: For github runners new windows are not supported so noMintty

**Comment by shaileshpandit** [comment]
You can approve workflow runs from public forks as described [here](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks) to see build run on this pull request.

**Comment by TheElixZammuto** [comment]
Thank you!

**Pull request merged**